chromium/background-scripts/rules.js

```diff
@@ -546,51 +572,80 @@ RuleSets.prototype = {
       util.log(util.DBUG, "Ruleset cache miss for " + host);
     }
 
-    // Let's begin search
-    // Copy the host targets so we don't modify them.
-    let results = (this.targets.has(host) ?
-      new Set([...this.targets.get(host)]) :
-      new Set());
+    let results;
+    if (this.wasm_rs) {
+      let pa = this.wasm_rs.potentially_applicable(host);
+      results = [...pa].map(ruleset => {
```

Comment:
`results` should be a `Set` instead of an `Array`. Otherwise, `cached_item.size` is `undefined` when there is a cache hit.

Reply:
I'll convert this to a `Set`.  Currently, `this.wasm_rs.potentially_applicable` returns a `Set`, but we have to call `map` on the result because WebAssembly does not have access to the `CookieRule` or `Rule` objects.  This may be solvable, in which case we won't have to run this `map` at all.  I'll also look into this as a longer-term solution.